game.js

```diff
@@ -113,6 +113,42 @@ class Game {
                 this.resumeGame();
             }
 
+            // the code below can be viewed as a maxperiods + maxperiodtime problem, where period=pause
+            // if maxpausetime and 
+            if ((config.maxpausetime || config.maxpauses) && clock.pause.paused && !clock.pause.pause_control["stone-removal"] && !clock.pause.pause_control.system && !clock.pause.pause_control.weekend && !clock.pause.pause_control["vacation-" + clock.black_player_id] && !clock.pause.pause_control["vacation-" + clock.white_player_id]) {
+
+                let pausesNumberConsumed = 0; // max is config.maxpauses
```

Comment:
Move this up to around line 25 or so. It should be in the initial variables of the constuctor, so that we can store it and access it later as part of this game object. It won't be recognized otherwise down inside the pausetimeCheck() callback, as I understand things. Someone else speak up if I'm wrong.

Reply:
You'd do like ```this.pausesNumberConsumed = 0;```

Reply:
indeed, i shouldnt put it in the function !!